### PR TITLE
Asset Discovery Skip Reason

### PR DESCRIPTION
**Pull request by djones**
Break out each reason we'd skip an asset from being discovered and output that to the log

### CODE REVIEW

File: src/services/response-service.ts [hunk 1]
```diff
@@ -31,15 +31,27 @@ export default class ResponseService extends PercyClientService {
...
+    }
+
+    if (!this.ALLOWED_RESPONSE_STATUSES.includes(response.status())) {
       // Only allow 2XX responses:
-      || !this.ALLOWED_RESPONSE_STATUSES.includes(response.status())
-      || !request.url().startsWith(rootUrl) // Disallow remote resource requests.
-      || !response.url().startsWith(rootUrl) // Disallow remote redirects.
-      ) {
-      logger.debug(`Skipping: ${response.url()}`)
+      logger.debug(`Skipping [disallowed_response_status]: ${response.url()}`)
```
Comment (anaulin) [inline src/services/response-service.ts]: In this case might be nice to log the response status we got.

Comment (Robdel12) [reply src/services/response-service.ts]: I was going to say the same thing :D

Comment (djones) [reply src/services/response-service.ts]: included with bf131e59a7898e6f3afbd4cae85196ed1e52b62e

File: src/services/response-service.ts [hunk 2]
```diff
@@ -31,15 +31,27 @@ export default class ResponseService extends PercyClientService {
     const parsedRootResourceUrl = new URL(rootResourceUrl)
     const rootUrl = `${parsedRootResourceUrl.protocol}//${parsedRootResourceUrl.host}`
 
-    if (
+    if (request.url() === rootResourceUrl) {
       // Always skip the root resource
-      request.url() === rootResourceUrl
+      logger.debug(`Skipping [is_root_resource]: ${response.url()}`)
```
Comment (timhaines) [inline src/services/response-service.ts]: Should this be request.url()?

File: src/services/response-service.ts [hunk 3]
```diff
@@ -31,15 +31,27 @@ export default class ResponseService extends PercyClientService {
...
-      || !response.url().startsWith(rootUrl) // Disallow remote redirects.
-      ) {
-      logger.debug(`Skipping: ${response.url()}`)
+      logger.debug(`Skipping [disallowed_response_status_${response.status()}]: ${response.url()}`)
+      return
+    }
+
+    if (!request.url().startsWith(rootUrl)) {
+      // Disallow remote resource requests.
+      logger.debug(`Skipping [is_remote_resource]: ${response.url()}`)
```
Comment (timhaines) [inline src/services/response-service.ts]: Should this be request.url() ?

File: src/services/response-service.ts (hunk 2 above)
Comment (djones) [reply src/services/response-service.ts]: Fixed with f0749262a8cc68828db9c0278aabf8ad1c1b8ee7

File: src/services/response-service.ts (hunk 3 above)
Comment (djones) [reply src/services/response-service.ts]: Fixed with f0749262a8cc68828db9c0278aabf8ad1c1b8ee7

**Pull request merged**